evcxr_repl/src/scan.rs

```diff
@@ -91,20 +91,36 @@ pub enum FragmentValidity {
 pub fn validate_source_fragment(source: &str) -> FragmentValidity {
     use Bracket::*;
     let mut stack: Vec<Bracket> = vec![];
+    // The expected depth `stack` should have after the closing ']' of the attribute
+    // is read; None if closing ']' has already been read or currently not reading
+    // attribute
+    let mut attr_end_stack_depth: Option<usize> = None;
+    // Whether the item after an attribute is expected; is set to true after the
+    // expected attr_end_stack_depth was reached
+    let mut expects_attr_item = false;
 
     let mut input = source.char_indices().peekable();
     while let Some((i, c)) = input.next() {
+        // For simplicity set `expects_attr_item = false` and only for all cases where next
```

Comment:
I guess one alternative to this would be to have a variable like: `let mut is_attr_target = true;`. Possible alternative names `is_code`, `non_empty`. Then instead of `expects_attr_item = expected_attr_item`, you can do `is_attr_target = false`. After the match you can then update `expects_attr_item`. e.g. `if is_attr_target {expects_attr_item = false}`.

I'll leave it up to you if you think this is more/less easy to understand. I'm happy to merge either way.

Reply:
Good idea, I am also not that happy with the `expects_attr_item = expected_attr_item` since it is a bit confusing. I have adjusted it in the latest commit, I hope that is what you had in mind.